Essentials/UI/CooldownButton.cs

```diff
@@ -1,295 +1,226 @@
                     if (hasEffectDuration)
                     {
-                        IsEffectActive = true;
-
+                        isEffectActive = true;
                         Timer = EffectDuration;
-
-                        KillButtonManager.TimerText.Color = new Color(0, 255, 0);
+                        killButtonManager.TimerText.Color = new Color(0, 255, 0);
                     }
-
-                    onClick();
+                    OnClick();
                 }
-            }));
-        }
-
-        /// <summary>
-        /// Call in HudManager.Start
-        /// </summary>
-        public CooldownButton(Action onClick, float cooldown, string imageEmbededResourcePath, Vector2 positionOffset, ButtonCategory category, float effectDuration, Action onEffectEnd) : this(Assembly.GetCallingAssembly(), onClick, cooldown, imageEmbededResourcePath, positionOffset, category, true, effectDuration, onEffectEnd)
-        {
+            }
         }
-
-        /// <summary>
-        /// Call in HudManager.Start
-        /// </summary>
-        public CooldownButton(Action onClick, float cooldown, string imageEmbededResourcePath, Vector2 positionOffset, ButtonCategory category) : this(Assembly.GetCallingAssembly(), onClick, cooldown, imageEmbededResourcePath, positionOffset, category, false, 0F, null)
+        public bool CanUse()
         {
+            if (PlayerControl.LocalPlayer.Data == null) return false;
+            canUse = useTester();
+            return true;
         }
-
-        public bool IsUsable()
+        public static void HudUpdate()
         {
-            if (PlayerControl.LocalPlayer?.Data == null) return false;
-
-            switch (Category)
+            buttons.RemoveAll(item => item.killButtonManager == null);
+            for (int i = 0; i < buttons.Count; i++)
             {
-                case ButtonCategory.Everyone:
-                {
-                    CanUse = true;
-
-                    break;
-                }
-                case ButtonCategory.OnlyCrewmate:
+                try
                 {
-                    CanUse = !PlayerControl.LocalPlayer.Data.IsImpostor;
-
-                    break;
-                }
-                case ButtonCategory.OnlyImpostor:
-                {
-                    CanUse = PlayerControl.LocalPlayer.Data.IsImpostor;
-
-                    break;
+                    if (buttons[i].CanUse())
+                        buttons[i].Update();
                 }
-            }
-
-            return true;
-        }
-
-        [HarmonyPatch(typeof(HudManager), nameof(HudManager.Update))]
-        private static class HudUpdatePatch
-        {
-            public static void Postfix()
-            {
-                Buttons.RemoveAll(item => item.KillButtonManager == null);
-
-                for (int i = 0; i < Buttons.Count; i++)
+                catch (NullReferenceException)
```

Comment:
Happens due to the improper `PlayerControl.LocalPlayer.Data` check mentioned above. Corrected in my refactor.